modules/lib/src/main/resources/assets/js/app/view/AccessControlEntryView.ts

```diff
@@ -136,37 +136,34 @@ export class AccessControlEntryView
     }
 
     private static canWrite(allowed: Permission[]): boolean {
-        return this.canRead(allowed) &&
-               allowed.indexOf(Permission.CREATE) >= 0 &&
+        return allowed.indexOf(Permission.CREATE) >= 0 &&
                allowed.indexOf(Permission.MODIFY) >= 0 &&
                allowed.indexOf(Permission.DELETE) >= 0;
     }
 
     private static canOnlyWrite(allowed: Permission[]): boolean {
-        return this.canWrite(allowed) && allowed.length === 4;
+        return this.canWrite(allowed) && allowed.length === 3;
     }
 
     private static canPublish(allowed: Permission[]): boolean {
-        return this.canWrite(allowed) &&
-               allowed.indexOf(Permission.PUBLISH) >= 0;
+        return allowed.indexOf(Permission.PUBLISH) >= 0;
     }
 
     private static canOnlyPublish(allowed: Permission[]): boolean {
-        return this.canPublish(allowed) && allowed.length === 5;
+        return this.canPublish(allowed) && allowed.length === 1;
     }
 
     private static isFullAccess(allowed: Permission[]): boolean {
-        return this.canPublish(allowed) &&
-               allowed.indexOf(Permission.READ_PERMISSIONS) >= 0 &&
+        return allowed.indexOf(Permission.READ_PERMISSIONS) >= 0 &&
                allowed.indexOf(Permission.WRITE_PERMISSIONS) >= 0;
     }
```

Comment:
Hmm, why should _only_ `READ_PERMISSIONS` + `WRITE_PERMISSIONS` mean `FULL ACCESS`? FA is when **all** available permissions are applied. In the current code it was achieved with `this.canPublish(allowed)` + these two, where `this.canPublish` is a composite check for the other 5.

Reply:
The thing is that getPermissionsValueFromAccess() considers acess is FULL when just these two READ_PERMISSIONS + WRITE_PERMISSIONS present

Reply:
Ah, I see now why - there was no `break;` before in `SWITCH` statement. Then yes, will fix in other way